protostar/starknet/cheatable_cached_state.py

```diff
@@ -174,6 +177,32 @@ def unregister_expected_call(
             if not data_for_address:
                 del self.expected_contract_calls[contract_address]
 
+    def register_expected_call_cairo(
```

Comment:
For now, I decided to keep both versions of functions for `expect_call` - the original ones and the new ones for the new arch.

Reply:
**TODO** Worth re-checking if there's a way to use the same structure (and registering functions) for both

Reply:
[unify expected call data](https://github.com/software-mansion/protostar/pull/1363/commits/dce29d380746999059899f807331f7f29dc849d5)